ext/date/tests/date_period_unserialize3.phpt

```diff
@@ -33,18 +33,25 @@ object(DatePeriod)#%d (%d) {
   ["start"]=>
   object(DateTime)#%d (%d) {
     ["date"]=>
-    string(26) "2022-07-14 00:00:00.000000"
+    string(26) "2024-08-28 00:00:00.000000"
```

Comment:
Unserialization doesn't revert the already done changes, therefore if an error happens at `include_start_date` (due to the wrong time) then the expected result is that the new value is applied for `current`, `start`, `end` etc.

Reply:
Isn't this a change in behaviour then? I don't quite understand why you need to make this changes.

Reply:
The previous behavior  was a by-product of storing properties in two places. The following happened: internal struct members of `DatePeriod` were assigned a value during unserialization, until an incompatible type of the `include_start_date` key caused a failure. That's why regular object properties were left intact, since https://github.com/php/php-src/pull/15598/files#diff-975ac482adc74b487c6ac9d8dde32153e4b8a803ef8ec41e8d6594f0a042db69L5846 was skipped. The test reflected the old object property values then, while the internal struct contained the new values.

The new behavior is consistent with the comment here: https://github.com/php/php-src/blob/176d28570030ab40457631be79c635f0ddbcfdd3/ext/date/php_date.c#L5694 (`/* this function does no rollback on error */`). If there is an error during serialization, there is no "rollback", so the test reflects the new values for those properties which were affected by the unserialization.